### PR TITLE
Chapter 9 - Working with Forms

### PR DESCRIPTION
**Pull request by ViolaRafael**
- Added option to manage products: edit existing products, remove products and create new products.
- Added option to prevent accidently removing products from Cart and Manage Products.

### CODE REVIEW

**Comment by PedroRaposoPMA** [comment]
The pull request description should be in English.

File: lib/components/app_drawer.dart [hunk 1]
```diff
@@ -10,25 +10,33 @@ class AppDrawer extends StatelessWidget {
       child: Column(
         children: [
           AppBar(
-            title: Text('Bem vindo Usuário!'),
+            title: const Text('Bem vindo Usuário!'),
```
Comment (PedroRaposoPMA) [inline lib/components/app_drawer.dart]: Use pt-pt (utilizador)

File: lib/components/cart_item.dart [hunk 2]
```diff
@@ -27,6 +27,29 @@ class CartItemWidget extends StatelessWidget {
           size: 40,
         ),
       ),
+      confirmDismiss: (_) {
+        return showDialog<bool>(
+          context: context,
+          builder: (ctx) => AlertDialog(
+            title: const Text('Tem Certeza?'),
+            content: const Text('Quer Remover o tem do carrinho?'),
```
Comment (PedroRaposoPMA) [inline lib/components/cart_item.dart]: Missing connector (o `que` tem)

File: lib/models/product_list.dart [hunk 3]
```diff
@@ -9,8 +10,45 @@ class ProductList with ChangeNotifier {
   List<Product> get favoriteItems =>
       _items.where((prod) => prod.isFavorite).toList();
 
+  int get itemsCount {
+    return _items.length;
+  }
+
+  void saveProduct(Map<String, Object> data) {
```
Comment (PedroRaposoPMA) [inline lib/models/product_list.dart:19]: Add documentation

File: lib/models/product_list.dart [hunk 4]
```diff
@@ -9,8 +10,45 @@ class ProductList with ChangeNotifier {
...
+  }
+
   void addProduct(Product product) {
     _items.add(product);
     notifyListeners();
   }
-}
\ No newline at end of file
+
+  void updateProduct(Product product) {
```
Comment (PedroRaposoPMA) [inline lib/models/product_list.dart:45]: Add documentation

File: lib/models/product_list.dart [hunk 5]
```diff
@@ -9,8 +10,45 @@ class ProductList with ChangeNotifier {
...
+    int index = _items.indexWhere((p) => p.id == product.id);
+    if(index >= 0) {
+      _items[index] = product;
+      notifyListeners();
+    }
+  }
+  void removeProduct(Product product) {
```
Comment (PedroRaposoPMA) [inline lib/models/product_list.dart:54]: Add documentation

**Review by PedroRaposoPMA — CHANGES_REQUESTED** [review]

File: lib/components/cart_item.dart (hunk 2 above)
Comment (samuelcpedro) [inline lib/components/cart_item.dart]: Update to 'Quer remover o item do carrinho?'

**Review by samuelcpedro — CHANGES_REQUESTED** [review]

**Review by PedroRaposoPMA — APPROVED** [review]

**Pull request merged**